aioresponses/core.py

```diff
@@ -94,33 +104,52 @@ def match(self, method: str, url: URL) -> bool:
             kwargs['writer'] = Mock()
             kwargs['continue100'] = None
             kwargs['timer'] = TimerNoop()
-            if AIOHTTP_VERSION >= StrictVersion('3.3.0'):
-                pass
-            else:
+            if AIOHTTP_VERSION < StrictVersion('3.3.0'):
                 kwargs['auto_decompress'] = True
             kwargs['traces'] = []
             kwargs['loop'] = loop
             kwargs['session'] = None
-        resp = self.response_class(self.method, url, **kwargs)
-        # we need to initialize headers manually
-        headers = CIMultiDict({hdrs.CONTENT_TYPE: self.content_type})
-        if self.headers:
-            headers.update(self.headers)
-        raw_headers = self._build_raw_headers(headers)
+        _headers = CIMultiDict({hdrs.CONTENT_TYPE: content_type})
+        if headers:
+            _headers.update(headers)
+        raw_headers = self._build_raw_headers(_headers)
+        resp = response_class(method, url, **kwargs)
         if AIOHTTP_VERSION >= StrictVersion('3.3.0'):
             # Reified attributes
-            resp._headers = headers
+            resp._headers = _headers
             resp._raw_headers = raw_headers
         else:
-            resp.headers = headers
+            resp.headers = _headers
             resp.raw_headers = raw_headers
-        resp.status = self.status
-        resp.reason = self.reason
+        resp.status = status
+        resp.reason = reason
         resp.content = stream_reader_factory()
-        resp.content.feed_data(self.body)
+        resp.content.feed_data(body)
         resp.content.feed_eof()
         return resp
 
+    async def _build_response(
+            self, url: URL, *args: Tuple, **kwargs: Dict
+    ) -> 'Union[ClientResponse, Exception]':
+        if isinstance(self.exception, Exception):
+            return self.exception
+        if self.callback and callable(self.callback):
+            resp = await self.callback(self, url, *args, **kwargs)
```

Comment:
Maybe callback annotation should look like
```
def callback(url: str, **kwargs):
```

for now we could forgot about `*args` argument, it's not used by aiohttp anyway :)

Reply:
Agree, removed it.